tests/webtest/test_UTF8InFilename.js

```diff
@@ -0,0 +1,36 @@
+add_task(async function test_UTF8InFilename_Search() {
+  await TestUtils.loadPath("/");
+  TestUtils.shortenSearchTimeouts();
+
+  const query = frame.contentDocument.querySelector("#query");
+  TestUtils.setText(query, "SymbolInFilenameWithUTF8");
+
+  const content = frame.contentDocument.querySelector("#content");
+
+  await waitForCondition(
+    () => content.textContent.includes("Core code (1 lines") &&
+      content.textContent.includes("Definitions (SymbolInFilenameWithUTF8) (1 lines") &&
+      content.textContent.includes("var SymbolInFilenameWithUTF8"),
+    "symbol in file with space in filename matches as definition");
+});
+
+add_task(async function test_UTF8InFilename_DirAndFile() {
+  await TestUtils.loadPath("/tests/source/js");
+
+  const link = frame.contentDocument.querySelector(`.folder-content a[href*="with-UTF8-"]`);
+  ok(!!link, "UTF-8 filename is listed");
+  is(link.getAttribute("href"), "/tests/source/js/with-UTF8-ファイル.js",
+     "UTF-8 path is used with raw text");
```

Comment:
Although raw text is working right now, we might want to URL-encode the path when embedding into HTML.

Reply:
It seems like these are [legal URL code points](https://url.spec.whatwg.org/#url-code-points) and so this might actually be the most readable thing we can do?  The chain of HTML spec definitions leading to that is:
- [The href attribute on a and area elements must have a value that is a valid URL potentially surrounded by spaces.](https://html.spec.whatwg.org/multipage/links.html#attr-hyperlink-href)
- [A string is a valid URL potentially surrounded by spaces if, after stripping leading and trailing ASCII whitespace from it, it is a valid URL string.](https://html.spec.whatwg.org/multipage/urls-and-fetching.html#valid-url-potentially-surrounded-by-spaces)
- https://url.spec.whatwg.org/#valid-url-string
- https://url.spec.whatwg.org/#absolute-url-with-fragment-string
- https://url.spec.whatwg.org/#absolute-url-string
- https://url.spec.whatwg.org/#scheme-relative-special-url-string
- https://url.spec.whatwg.org/#path-absolute-url-string
- https://url.spec.whatwg.org/#path-relative-url-string
- https://url.spec.whatwg.org/#url-path-segment-string
- https://url.spec.whatwg.org/#url-units
- https://url.spec.whatwg.org/#url-code-points

Reply:
oh, right, but maybe we would want to encode the spaces, which is maybe what you meant

Reply:
Also, I should say, I'm not opposed to just encoding the URLs, but I often will check where URLs go in the Firefox URL target hover thing in the lower left, and obviously a fully encoded URL is much more inscrutable than the non-encoded version.